scrapers/csci_topics_scraper/main.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+
+# Python standard library
+import asyncio
+from operator import itemgetter
+import os
+import re
+import json
+import sys
+
+# External dependnecies
+import aiohttp
+import bs4
+import requests
+
+BeautifulSoup = lambda data: bs4.BeautifulSoup(data, features="lxml")
+
+
+async def get_topics_txts():
+    global session
+    url = "https://www.cs.rpi.edu/~goldsd/docs/"
+    async with session.get(url) as request:
+        soup = BeautifulSoup(await request.text())
+        links = soup.find("table").findAll("a")
+        out = []
+        for link in links:
+            if "topics-courses" in link["href"]:
+                out.append(link["href"])
+        return out
+
+
+async def scrape_all_txts(links):
+    list_of_terms = {}
+    for link in links:
+        await scrape_txt(link, list_of_terms)
+    return list_of_terms
+
+
+def goldy_term_to_sis_term(term):
+    term = term.lower()
+    if term[0] == "s":
+        postfix = "01"
+    elif term[0] == "u":
+        postfix = "05"
+    elif term[0] == "f":
+        postfix = "09"
+    else:
+        # placeholder
+        postfix = "00"
+    return "20" + term[1:] + postfix
+
+
+async def scrape_txt(filename, list_of_terms):
+    global session
+    url = f"https://www.cs.rpi.edu/~goldsd/docs/{filename}"
+    async with session.get(url) as request:
+        data = re.split(r"\n======*", (await request.text()))
+        for term_data in data:
+            courses = [
+                line
+                for line in re.split("\n\n", term_data.replace("\r", ""))
+                if line.strip()
+            ]
+            term = courses[0].replace("[", "").split(" ")[0]
+            list_of_terms[goldy_term_to_sis_term(term)] = scrape_term(courses[1:])
+        return list_of_terms
+
+
+def scrape_term(courses):
```

Comment:
Nit: this should be `parse_term` or something similar as it's not scraping anything new.

Reply:
All right, sounds good